src/Service/Terminal/Service_Terminal.re

```diff
@@ -67,22 +67,30 @@ module Sub = {
         let makeDebouncedDispatch = () => {
```

Comment:
It would also be nice if this was generic. I see no reason it needs to be hard-coded for `dispatch`.

Reply:
True. I refactored to a `FunEx.debounce1` method

Reply:
Very nice. I'm not sure about just calling it `debounce1` though, it has a subtly different behaviour than a "normal" debounce. Perhaps `debounceLater1`? Or have an optional  `immediate` argument like [underscore](http://underscorejs.org/#debounce) does.

Reply:
Ah, you're right... Looking at the debounce docs, it's more akin to `throttle` because we don't reset the timer on each invocation (like `debounce` does). I switched the name, and added arguments for `~leading` and `~trailing` behavior.